model/src/main/proto/feedback_reporting.proto

```diff
@@ -0,0 +1,136 @@
+syntax = "proto3";
+
+package model;
+
+import "profile.proto";
+
+option java_package = "org.oppia.android.app.model";
+option java_multiple_files = true;
+
+// Top-level proto to store feedback reports locally when there is no network connection to upload
+// them to the remote service
+message FeedbackReportingDatabase {
+  repeated FeedbackReport reports_list = 1;
+}
+
+// Corresponds to an entire feedback report with user reported information and collected app,
+// device, and build information.
+message FeedbackReport {
+  // The date the report was created.
+  int64 report_date_ms = 1;
+  FeedbackReportingForm feedback_form = 2;
+  FeedbackReportingDeviceInfo device_info = 3;
+  FeedbackReportingBuildInfo build_info = 4;
+  FeedbackReportingAppInfo app_info = 5;
+}
+
+// Information supplied by the user in the feedback report.
+message FeedbackReportingForm {
+  oneof report_type {
+    Suggestion suggestion = 1;
+    Issue issue = 2;
+    Crash crash = 3;
+  }
+}
+
+// A suggestion feedback report.
```

Comment:
Or perhaps: ``User-provided suggestion for something that could or should change in the app``. Consider that:
1. Readers likely have much less context than you (which will be true for as long as this code exists)
2. We want to establish 'API contracts' rather than just explain what the current behavior is since this code can & will evolve over time, and we should try to make it clear where certain changes should go

Ditto for other proto messages here.

Reply:
Done, ack!